packages/twenty-server/src/workspace/workspace-sync-metadata/factories/feature-flags.factory.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { Repository } from 'typeorm';
+
+import { WorkspaceSyncContext } from 'src/workspace/workspace-sync-metadata/interfaces/workspace-sync-context.interface';
+import { FeatureFlagMap } from 'src/core/feature-flag/interfaces/feature-flag-map.interface';
+
+import { FeatureFlagEntity } from 'src/core/feature-flag/feature-flag.entity';
+
+@Injectable()
+export class FeatureFlagFactory {
```

Comment:
In my mind, factories can be used when the create or modify an object based on other object, and we're creating a map of feature flags [key: value], it's why I've created a factory.
But if we think it can be used in other part of the code we can move it into `FeatureFlagService`

Reply:
To your last point: I think we will want this function in other places. Not fully sure though since usually we want to query over a specific featureFlag key and we don't need all of them but could be useful I guess 🙂

Reply:
I guess we can move it later if we need it ?